pkg/telemetry/phonehome/client.go

```diff
@@ -0,0 +1,286 @@
+package phonehome
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/grpc/authn"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome/segment"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome/telemeter"
+	"github.com/stackrox/rox/pkg/version"
+	"google.golang.org/grpc"
+)
+
+// Client wraps telemetry configuration and implements some related methods.
+type Client struct {
+	config   Config
+	stateMux sync.RWMutex
+
+	telemeter     telemeter.Telemeter
+	onceTelemeter sync.Once
+
+	gatherer     Gatherer
+	onceGatherer sync.Once
+
+	// Map of event name to the list of interceptors, that gather properties for
+	// the event.
+	interceptors     map[string][]Interceptor
+	interceptorsLock sync.RWMutex
+
+	// enabled is an additional switch to enable or disable a well configured
+	// client.
+	enabled bool
+}
+
+// NewClient returns a configured client instance.
+func NewClient(cfg *Config) *Client {
+	if cfg == nil ||
+		cfg.StorageKey == DisabledKey ||
+		// Disable telemetry when running debug version or unit tests if no key
+		// is configured.
+		cfg.StorageKey == "" && (!version.IsReleaseVersion() || testing.Testing()) {
+		return &Client{config: Config{StorageKey: DisabledKey}}
+	}
```

Comment:
```suggestion
	if cfg == nil {
		return &Client{config: Config{StorageKey: DisabledKey}}
	}

	// Disable telemetry to avoid reporting in non-production environments.
	if cfg.StorageKey == "" && (!version.IsReleaseVersion() || testing.Testing()) {
		cfg.StorageKey = DisabledKey
	}
```